erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -482,13 +482,13 @@ def set_basic_rate(self, reset_outgoing_rate=True, raise_error_if_no_rate=True):
 			d.basic_rate = flt(d.basic_rate, d.precision("basic_rate"))
 			d.basic_amount = flt(flt(d.transfer_qty) * flt(d.basic_rate), d.precision("basic_amount"))
 
-	def set_rate_for_outgoing_items(self, reset_outgoing_rate=True):
+	def set_rate_for_outgoing_items(self, reset_outgoing_rate=True, raise_error_if_no_rate=True):
 		outgoing_items_cost = 0.0
 		for d in self.get('items'):
 			if d.s_warehouse:
 				if reset_outgoing_rate:
 					args = self.get_args_for_incoming_rate(d)
-					rate = get_incoming_rate(args)
+					rate = get_incoming_rate(args, raise_error_if_no_rate)
```

Comment:
Although this solves a problem, it does only for this one case.
Since **Allow zero valuation rate** is not auto-set yet, to this function, right now, it is an ordinary stock item that should have some valuation rate in the system.
In most cases it will be an ordinary item (not customer provided), for which raising the error is important. The valuation rate should be set at this stage.

With your fix it just does not raise the error even if a **non-customer-provided** item is transacting without a valuation rate. 

It would be best to just not calculate the rate for customer provided items in the Stock Entry and set it to 0 blindly. But there's no way to know if it is a customer provided item in the Items table.

>Alternatively, the validate_customer_provided_item method which sets the Allow Zero Valuation Rate field can be run before calculate_rate_and_amount is called.

This seems like the best and cleanest idea.